Shared/Samples/Validate utility network topology/ValidateUtilityNetworkTopologyView.Model.swift

```diff
@@ -0,0 +1,448 @@
+        @Published var statusMessage = ""
+        
+        /// A Boolean value indicating whether the current state of the utility network can be obtained.
+        @Published private(set) var canGetState = false
+        
+        /// A Boolean value indicating whether a trace can be run.
+        @Published private(set) var canTrace = false
+        
+        /// A Boolean value indicating whether the utility network topology can be validated.
+        @Published private(set) var canValidateNetworkTopology = false
+        
+        /// A Boolean value indicating whether there is a selection that can be cleared.
+        @Published private(set) var canClearSelection = false
+        
+        deinit {
+            ArcGISEnvironment.authenticationManager.arcGISCredentialStore.removeAll()
+        }
+        
+        // MARK: Methods
+        
+        /// Gets the current state of the utility network and updates the status with the results.
+        func getState() async throws {
+            statusMessage = "Getting utility network state..."
+            
+            // Get the current state of the utility network.
+            let state = try await utilityNetwork.state
+            
+            // Allow validating if the network contains any dirty areas or errors.
+            canValidateNetworkTopology = state.hasDirtyAreas || state.hasErrors
+            
+            // Allow tracing if the network has topology is enabled.
+            canTrace = state.networkTopologyIsEnabled
+            
+            // Update the status with the state.
+            let instructionMessage = canValidateNetworkTopology
+            ? "Tap 'Validate' before trace or expect a trace error."
+            : "Tap on a feature to edit or tap 'Trace' to run a trace."
+            
+            statusMessage = """
+            Utility Network State:
+            Has dirty areas: \(state.hasDirtyAreas)
+            Has errors: \(state.hasErrors)
+            Network topology is enabled: \(state.networkTopologyIsEnabled)
+            \(instructionMessage)
+            """
+        }
+        
+        /// Runs a trace and selects features in the map that correspond to the resulting elements.
+        func trace() async throws {
+            statusMessage = "Running a downstream trace..."
+            clearLayerSelections()
+            
+            // Get the element trace result from the utility network using the trace parameters.
+            let traceResults = try await utilityNetwork.trace(using: traceParameters)
+            let elementTraceResult = traceResults.first {
+                $0 is UtilityElementTraceResult
+            } as! UtilityElementTraceResult
+            
+            // Select all of elements found.
+            statusMessage = "Selecting found elements..."
+            
+            guard let layers = map.operationalLayers as? [FeatureLayer] else { return }
+            for layer in layers {
```

Comment:
it would be better to filter by feature layers than to cast the whole array. If somebody copies this code and they have other layer types in there it won't as is. Best to be a bit more lenient here.
```suggestion
            for layer in map.operationalLayers.compactMap({ $0 as? FeatureLayer }) {
```